sw/host/opentitanlib/src/util/status.rs

```diff
@@ -0,0 +1,255 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+// This utility handles the various aspect of the status_t type used in
+// the device code.
+
+use anyhow::{bail, Context, Result};
+use bindgen::status::{ot_status_create_record_t, status_create, status_err, status_extract};
+
+use object::{Object, ObjectSection};
+
+use num_enum::TryFromPrimitive;
+
+use std::convert::TryFrom;
+use std::ffi::CString;
+use std::path::PathBuf;
+
+pub use bindgen::status::absl_status_t as RawStatusCode;
+pub use bindgen::status::status_t as RawStatus;
+
+// FIXME: is there a better way of doing this? bindgen does not
```

Comment:
In Rust-for-Linux we postprocess rs files generated by bindgen with sed.

Reply:
FYI, bindgen does support custom derives but not through the CLI.

https://docs.rs/bindgen/latest/bindgen/callbacks/trait.ParseCallbacks.html#method.add_derives